### PR TITLE
Add shared GHA for ruby

### PR DESCRIPTION
**Pull request by girijaasoni**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,8 @@ source "https://rubygems.org"
 
 gemspec
 
+gem 'hammer_cli_foreman', github: 'theforeman/hammer-cli-foreman', branch: 'master'
```
Comment (archanaserver) [inline Gemfile]: I wonder why do we need this here, when we already have this dependency in .gemspec file?

File: hammer_cli_foreman_bootdisk.gemspec [hunk 2]
```diff
@@ -19,6 +19,7 @@ Gem::Specification.new do |s|
   s.extra_rdoc_files = Dir['LICENSE', 'README*']
   s.require_paths = ["lib"]
 
-  s.add_dependency 'hammer_cli_foreman', '>= 0.1.2'
```
Comment (ofedoren) [inline hammer_cli_foreman_bootdisk.gemspec:22]: Please, don't pin hammer core version like that. We mostly use "minimal" version without any upper limit. In worst cases we limit up to next major version, e.g. 4.0.0.

File: Gemfile [hunk 3]
```diff
@@ -3,10 +3,19 @@ source "https://rubygems.org"
...
 
 gem 'gettext', '>= 3.1.3', '< 4.0.0'
-gem 'rake', '~> 13.1.0'
+
+group :test do
+  gem 'ci_reporter', '>= 1.6.3', '< 2.0.0', :require => false
+  gem 'minitest'
+  gem 'minitest-spec-context'
+  gem 'mocha'
+  gem 'rake', '~> 13.0'
```
Comment (ofedoren) [inline Gemfile]: I don't think we need to pin rake. If tests are broken, we should fix that. If there's a breaking change in the lib, we should adjust.

**Review by ofedoren — COMMENTED** [review]
I've tried to push this commit on my fork and create a PR here, it doesn't trigger the GHA as well. I think GHA workflow file should be present in master and then we can trigger it with different PR. Do I understand it right, @ekohl (sorry for pinging you again regarding GHA :/ )?

Also, hammer plugins used to be tested against latest master branch of hammer-cli[-foreman], but AFAIU shared `test-gem` workflow simply installs the gem with dependencies (such as particular/latest released version of a dependency) in .gemspec/Gemfile and triggers its tests.

I'm not sure if it's sufficient. I mean, it's great that it will run the tests against already released version, but then we won't be able to see through CI if the latest changes of unreleased version caused some regressions. The only way to know that will be old good checkout with Gemfile.local and running the tests locally.

If I was adding something similar that we have now in Foreman, I'd rather create a separate workflow like `hammer_plugin` instead of `test-gem`.

Also some comments inline.

File: Rakefile [hunk 4]
```diff
@@ -3,9 +3,20 @@ require 'bundler/gem_tasks'
 require 'hammer_cli/i18n/find_task'
 require_relative "./lib/hammer_cli_foreman_bootdisk/version"
 require_relative "./lib/hammer_cli_foreman_bootdisk/i18n"
+require 'ci/reporter/rake/minitest'
```
Comment (nofaralfasi) [inline Rakefile]: Do we need this even though there are no tests in this repo?

File: Rakefile [hunk 5]
```diff
@@ -3,9 +3,20 @@ require 'bundler/gem_tasks'
...
 
 HammerCLI::I18n::FindTask.define(HammerCLIForemanBootdisk::I18n::LocaleDomain.new, HammerCLIForemanBootdisk.version)
 
+
+Rake::TestTask.new do |t|
+  t.libs.push 'lib'
+  t.test_files = Dir.glob('test/**/*_test.rb')
```
Comment (nofaralfasi) [inline Rakefile]: Same here as above

**Comment by ekohl** [comment]
> I've tried to push this commit on my fork and create a PR here, it doesn't trigger the GHA as well. I think GHA workflow file should be present in master and then we can trigger it with different PR. Do I understand it right, @ekohl (sorry for pinging you again regarding GHA :/ )?

I added a comment to describe it: https://community.theforeman.org/t/convert-foreman-plugins-continuous-integration-to-github-actions/35917/7

Now it may also need to be triggered once, so you may need to modify the triggers (temporarily).

**Comment by ofedoren** [comment]
Hmm... first time I see that open-ended dependency for a gem causes fails...

**Comment by ekohl** [comment]
Building a gem gives warnings about it and validating it makes those warnings errors. Perhaps you can teach it to be more tolerant but I generally think it's a good idea.

**Comment by girijaasoni** [comment]
> >so you may need to modify the triggers (temporarily).

It doesn't have the test folder, so I dropped the test-gem 
updated just the rubocop for the style enforcement. Would that be enough? @ekohl @ofedoren

File: .github/workflows/ruby.yml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
+
```
Comment (nofaralfasi) [inline .github/workflows/ruby.yml]:
```suggestion

```

File: .rubocop.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  Enabled: false
+
+Style/FrozenStringLiteralComment:
+  Enabled: false
+
+Gemspec/RequireMFA:
+  Enabled: false
+
+Style/SymbolProc:
+  Enabled: false
```
Comment (nofaralfasi) [inline .rubocop.yml]: Missing a new line here

File: Gemfile [hunk 8]
```diff
@@ -1,12 +1,19 @@
-source "https://rubygems.org"
+source 'https://rubygems.org'
 
 gemspec
 
 gem 'gettext', '>= 3.1.3', '< 4.0.0'
 gem 'rake', '~> 13.1.0'
 
+gem 'hammer_cli_foreman', github: 'theforeman/hammer-cli-foreman', branch: 'master'
```
Comment (nofaralfasi) [inline Gemfile]: Is this necessary?

File: hammer_cli_foreman_bootdisk.gemspec [hunk 9]
```diff
@@ -12,13 +10,13 @@ Gem::Specification.new do |s|
...
-  s.summary       = %q{Foreman boot disk commands for Hammer}
-  s.description   = %q{Foreman boot disk commands for Hammer CLI}
+  s.summary       = 'Foreman boot disk commands for Hammer'
+  s.description   = 'Foreman boot disk commands for Hammer CLI'
 
   s.files            = Dir['{config,lib,locale}/**/*', 'LICENSE', 'README*']
   s.extra_rdoc_files = Dir['LICENSE', 'README*']
   s.require_paths = ["lib"]
 
   s.add_dependency 'hammer_cli_foreman', '>= 0.1.2'
```
Comment (nofaralfasi) [inline hammer_cli_foreman_bootdisk.gemspec]:
```suggestion
  s.add_dependency 'hammer_cli_foreman', '~> 3.10'
```

File: Gemfile (hunk 8 above)
Comment (girijaasoni) [reply Gemfile]: Not with the updated version :D

File: .github/workflows/ruby.yml [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+      - 'master'
+
+concurrency:
+  group: ${{ github.ref_name }}-${{ github.workflow }}
+  cancel-in-progress: true
+
+jobs:
+  rubocop:
+    name: Rubocop
+    uses: theforeman/actions/.github/workflows/rubocop.yml@v0
```
Comment (nofaralfasi) [inline .github/workflows/ruby.yml]: Missing a new line (with no spaces).

File: .rubocop.yml [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (nofaralfasi) [inline .rubocop.yml]: Same here - the file should end with an empty line (no spaces).

**Review by stejskalleos — APPROVED** [review]

**Pull request merged**